File: win-prog/ProgWinCS/Buttons and Labels and Scrolls (Oh, My!)/TwoButtonsDock/TwoButtonsDock.cs
//---------------------------------------------
// TwoButtonsDock.cs © 2001 by Charles Petzold
//---------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class TwoButtonsDock: Form
{
     public static void Main()
     {
          Application.Run(new TwoButtonsDock());
     }
     public TwoButtonsDock()
     {
          Text = "Two Buttons with Dock";
          ResizeRedraw = true;

          Button btn = new Button();
          btn.Parent = this;
          btn.Text   = "&Larger";
          btn.Height = 2 * Font.Height;
          btn.Dock   = DockStyle.Top;
          btn.Click += new EventHandler(ButtonLargerOnClick);
          
          btn = new Button();
          btn.Parent = this;
          btn.Text   = "&Smaller";
          btn.Height = 2 * Font.Height;
          btn.Dock   = DockStyle.Bottom;
          btn.Click += new EventHandler(ButtonSmallerOnClick);
     }
     void ButtonLargerOnClick(object obj, EventArgs ea)
     {
          Left   -= (int)(0.05 * Width);
          Top    -= (int)(0.05 * Height);
          Width  += (int)(0.10 * Width);
          Height += (int)(0.10 * Height);
     }
     void ButtonSmallerOnClick(object obj, EventArgs ea)
     {
          Left   += (int)(Width  / 22f);
          Top    += (int)(Height / 22f);
          Width  -= (int)(Width  / 11f);
          Height -= (int)(Height / 11f);
     }
}

File: win-prog/ProgWinVB/Buttons and Labels and Scrolls (Oh, My!)/TwoButtonsDock/TwoButtonsDock.vb
'-----------------------------------------------
' TwoButtonsDock.vb (c) 2002 by Charles Petzold
'-----------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class TwoButtonsDock
    Inherits Form

    Shared Sub Main()
        Application.Run(New TwoButtonsDock())
    End Sub

    Sub New()
        Text = "Two Buttons with Dock"
        ResizeRedraw = True

        Dim btn As New Button()
        btn.Parent = Me
        btn.Text = "&Larger"
        btn.Height = 2 * Font.Height
        btn.Dock = DockStyle.Top
        AddHandler btn.Click, AddressOf ButtonLargerOnClick

        btn = New Button()
        btn.Parent = Me
        btn.Text = "&Smaller"
        btn.Height = 2 * Font.Height
        btn.Dock = DockStyle.Bottom
        AddHandler btn.Click, AddressOf ButtonSmallerOnClick
    End Sub

    Private Sub ButtonLargerOnClick(ByVal obj As Object, ByVal ea As EventArgs)
        Left -= CInt(0.05 * Width)
        Top -= CInt(0.05 * Height)
        Width += CInt(0.1 * Width)
        Height += CInt(0.1 * Height)
    End Sub

    Private Sub ButtonSmallerOnClick(ByVal obj As Object, ByVal ea As EventArgs)
        Left += CInt(Width / 22)
        Top += CInt(Height / 22)
        Width -= CInt(Width / 11)
        Height -= CInt(Height / 11)
    End Sub
End Class
